File: Aritmetica2/Aritmetica.java
public class Aritmetica {
    
    public int sumar(int a,int b) {
        return a+b;
    }
    public float sumar(float a,float b) {
        return a+b;
    }
    public float sumar(int a,float b) {
        return a+b;
    }
    public float sumar(float a,int b) {
        return a+b;
    }
   
    public int restar(int a,int b) {
        return a-b;
    }
    public float restar(float a,float b) {
        return a-b;
    }
    public float restar(int a,float b) {
        return a-b;
    }
    public float restar(float a,int b) {
        return a-b;
    }
    
    public int multiplicar(int a,int b) {
        return a*b;
    }
    public float multiplicar(float a,float b) {
        return a*b;
    }
    public float multiplicar(int a,float b) {
        return a*b;
    }
    public float multiplicar(float a,int b) {
        return a*b;
    }
    
    public int dividir(int a,int b) {
        return a/b;
    }
    public float dividir(float a,float b) {
        return a/b;
    }
    public float dividir(int a,float b) {
        return a/b;
    }
    public float dividir(float a,int b) {
       float r;
        if(b!=0){
           r= a/b;
        }
        else{r=0.0;}
        return r;
    }

    public static void main(String[] args) {
        Aritmetica a=new Aritmetica();
        
        System.out.println("   3 + 6   = "+a.sumar(3,6));
        System.out.println(" 3.5 + 6   = "+a.sumar(3.5f,6));
        System.out.println("   3 + 6.5 = "+a.sumar(3,6.5f));
        System.out.println(" 3.5 + 6.5 = "+a.sumar(3.5f,6.5f));

        System.out.println("  3 - 6    = "+a.restar(3,6));
        System.out.println("3.5 - 6    = "+a.restar(3.5f,6));
        System.out.println("  3 - 6.5  = "+a.restar(3,6.5f));
        System.out.println("3.5 - 6.5  = "+a.restar(3.5f,6.5f));

        System.out.println("  3 * 6    = "+a.multiplicar(3,6));
        System.out.println("3.5 * 6    = "+a.multiplicar(3.5f,6));
        System.out.println("  3 * 6.5  = "+a.multiplicar(3,6.5f));
        System.out.println("3.5 * 6.5  = "+a.multiplicar(3.5f,6.5f));

        System.out.println("  3 / 6    = "+a.dividir(3,6));
        System.out.println("3.5 / 6    = "+a.dividir(3.5f,6));
        System.out.println("  3 / 6.5  = "+a.dividir(3,6.5f));
        System.out.println("3.5 / 6.5  = "+a.dividir(3.5f,6.5f));
    }
   
}

File: Aritmetica2/Aritmetica.php
<?php

class Aritmetica {
    
    public function sumar() {
        $arg = func_get_args();
        $c=0;
        if(is_int($arg[0])and(is_int($arg[1]))){
            $c=(int)$arg[0]+(int)$arg[1];
        }
        if(is_float($arg[0])and(is_int($arg[1]))){
            $c=(float)$arg[0]+(int)$arg[1];
        }
        if(is_int($arg[0])and(is_float($arg[1]))){
            $c=(int)$arg[0]+(float)$arg[1];
        }
        if(is_float($arg[0])and(is_float($arg[1]))){
            $c=(float)$arg[0]+(float)$arg[1];
        }           
        return $c;
    }
    
    public function restar() {
        $arg = func_get_args();
        $c=0;
        if(is_int($arg[0])and(is_int($arg[1]))){
            $c=(int)$arg[0]-(int)$arg[1];
        }
        if(is_float($arg[0])and(is_int($arg[1]))){
            $c=(float)$arg[0]-(int)$arg[1];
        }
        if(is_int($arg[0])and(is_float($arg[1]))){
            $c=(int)$arg[0]-(float)$arg[1];
        }
        if(is_float($arg[0])and(is_float($arg[1]))){
            $c=(float)$arg[0]-(float)$arg[1];
        }           
        return $c;
    }

    public function multiplicar() {
        $arg = func_get_args();
        $c=0;
        if(is_int($arg[0])and(is_int($arg[1]))){
            $c=(int)$arg[0]*(int)$arg[1];
        }
        if(is_float($arg[0])and(is_int($arg[1]))){
            $c=(float)$arg[0]*(int)$arg[1];
        }
        if(is_int($arg[0])and(is_float($arg[1]))){
            $c=(int)$arg[0]*(float)$arg[1];
        }
        if(is_float($arg[0])and(is_float($arg[1]))){
            $c=(float)$arg[0]*(float)$arg[1];
        }           
        return $c;
    }

    public function dividir() {
        $arg = func_get_args();
        $c=0;
        if($arg[1]!=0){
            if(is_int($arg[0])and(is_int($arg[1]))){
                $c=(int)$arg[0]/(int)$arg[1];
            }
            if(is_float($arg[0])and(is_int($arg[1]))){
                $c=(float)$arg[0]/(int)$arg[1];
            }
            if(is_int($arg[0])and(is_float($arg[1]))){
                $c=(int)$arg[0]/(float)$arg[1];
            }
            if(is_float($arg[0])and(is_float($arg[1]))){
                $c=(float)$arg[0]/(float)$arg[1];
            }
        }
        else{$c="Error de division por cero";}     
        return $c;
    }
   
}
 $A= new Aritmetica();
 echo "   3 + 6   = ".$A->sumar(3,6)."<br>";
 echo " 3.5 + 6   = ".$A->sumar(3.5,6)."<br>";
 echo "   3 + 6.5 = ".$A->sumar(3,6.5)."<br>";
 echo " 3.5 + 6.5 = ".$A->sumar(3.5,6.5)."<br>";

 echo "  3 - 6    = ".$A->restar(3,6)."<br>";
 echo "3.5 - 6    = ".$A->restar(3.5,6)."<br>";
 echo "  3 - 6.5  = ".$A->restar(3,6.5)."<br>";
 echo "3.5 - 6.5  = ".$A->restar(3.5,6.5)."<br>";

 echo "  3 * 6    = ".$A->multiplicar(3,6)."<br>";
 echo "3.5 * 6    = ".$A->multiplicar(3.5,6)."<br>";
 echo "  3 * 6.5  = ".$A->multiplicar(3,6.5)."<br>";
 echo "3.5 * 6.5  = ".$A->multiplicar(3.5,6.5)."<br>";

 echo "  3 / 6    = ".$A->dividir(3,6)."<br>";
 echo "3.5 / 6    = ".$A->dividir(3.5,6)."<br>";
 echo "  3 / 6.5  = ".$A->dividir(3,6.5)."<br>";
 echo "3.5 / 6.5  = ".$A->dividir(3.5,6.5)."<br>";

 echo "3.5 / 0  = ".$A->dividir(3.5,0)."<br>";
?>